src/main/java/org/kohsuke/github/HttpException.java

```diff
@@ -0,0 +1,119 @@
+package org.kohsuke.github;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Http exception
+ *
+ * @author <a href="mailto:cleclerc@cloudbees.com">Cyrille Le Clerc</a>
+ */
+public class HttpException extends IOException {
+    static final long serialVersionUID = 1L;
+
+    private final int responseCode;
+    private final String responseMessage;
+    private final String url;
+
+    /**
+     * @param message         The detail message (which is saved for later retrieval
+     *                        by the {@link #getMessage()} method)
+     * @param responseCode    Http response code. {@code -1} if no code can be discerned.
+     * @param responseMessage Http response message
+     * @param url             The url that was invoked
+     * @see HttpURLConnection#getResponseCode()
+     * @see HttpURLConnection#getResponseMessage()
+     */
+    public HttpException(String message, int responseCode, String responseMessage, String url) {
+        super(message);
+        this.responseCode = responseCode;
+        this.responseMessage = responseMessage;
+        this.url = url;
+    }
+
+    /**
+     * @param message         The detail message (which is saved for later retrieval
+     *                        by the {@link #getMessage()} method)
+     * @param responseCode    Http response code. {@code -1} if no code can be discerned.
+     * @param responseMessage Http response message
+     * @param url             The url that was invoked
+     * @param cause           The cause (which is saved for later retrieval by the
+     *                        {@link #getCause()} method).  (A null value is permitted,
+     *                        and indicates that the cause is nonexistent or unknown.)
+     * @see HttpURLConnection#getResponseCode()
+     * @see HttpURLConnection#getResponseMessage()
+     */
+    public HttpException(String message, int responseCode, String responseMessage, String url, Throwable cause) {
+        super(message, cause);
+        this.responseCode = responseCode;
+        this.responseMessage = responseMessage;
+        this.url = url;
+    }
+
+    /**
+     * @param message         The detail message (which is saved for later retrieval
+     *                        by the {@link #getMessage()} method)
+     * @param responseCode    Http response code. {@code -1} if no code can be discerned.
+     * @param responseMessage Http response message
+     * @param url             The url that was invoked
```

Comment:
You need to ask the owner of this codebase what one they prefer.

Now as to which is correct:

* you are only allowed to use the `javax` package for authoring classes that you redistribute where the spec of those classes has been published via the JCP process.

* the JSR that was working on annotations to use for eg null status has not published anything at all. The only public statement is that they might introduce a `javax.annotations.NonNull` (which is not the one everyone is using you will note)

* as such we are breaking the good faith of the EULA or JCP etc agreements that maybe could - at an extreme stretch - be used to initiate proceedings... I doubt such proceedings would win... But then look at the Google - Oracle API appeal

So *where I have a choice* I will use the `edu.` version... under the non GPL license... Because in those cases I can choose to "do the right thing"... If there is ever even an official published draft of annotations for this in 'javax` I will switch to them